src/django/api/views/facility/facility_list_view_set.py

```diff
@@ -225,6 +282,215 @@ def create(self, request):
+            name = request.data['name']
+        else:
+            name = os.path.splitext(uploaded_file.name)[0]
+
+        if '|' in name:
+            raise ValidationError('Name cannot contain the "|" character.')
+
+        if 'description' in request.data:
+            description = request.data['description']
+        else:
+            description = None
+
+        if description is not None and '|' in description:
+            raise ValidationError(
+                'Description cannot contain the "|" character.'
+            )
+
+        replaces = None
+        if 'replaces' in request.data:
+            try:
+                replaces = int(request.data['replaces'])
+            except ValueError as exc:
+                raise ValidationError(
+                    '"replaces" must be an integer ID.'
+                ) from exc
+            old_list_qs = FacilityList.objects.filter(
+                source__contributor=contributor, pk=replaces)
+            if old_list_qs.count() == 0:
+                raise ValidationError(
+                    f'{replaces} is not a valid FacilityList ID.'
+                )
+            replaces = old_list_qs[0]
+            if FacilityList.objects.filter(replaces=replaces).count() > 0:
+                raise ValidationError(
+                    f'FacilityList {replaces.pk} has already been replaced.'
+                )
+
+        parsing_started = str(timezone.now())
+
+        serializer = self.__get_serializer(uploaded_file)
+        try:
+            rows = serializer.get_validated_rows()
+        except ValidationError as err:
+            report_error_to_rollbar(request=request, file=uploaded_file)
+            raise ValidationError(str(err.detail[0]))
+
+        if not self.__is_required_fields_present(rows):
+            raise ValidationError((
+                    f'Header must contain {FileHeaderField.COUNTRY}, '
+                    f'{FileHeaderField.NAME}, and '
+                    f'{FileHeaderField.ADDRESS} fields.'
+                ))
+
+        header_row_keys = rows[0].raw_json.keys()
+        header_str = ','.join(header_row_keys)
+        new_list = FacilityList(
+                    name=name,
+                    description=description,
+                    file_name=uploaded_file.name,
+                    file=uploaded_file,
+                    header=header_str,
+                    replaces=replaces,
+                    match_responsibility=contributor.match_responsibility)
+        new_list.save()
+
+        create_nonstandard_fields(header_row_keys, contributor)
+
+        source = Source.objects.create(
+            contributor=contributor,
+            source_type=Source.LIST,
+            facility_list=new_list)
+
+        is_geocoded = False
+        parsed_items = set()
+        for idx, row in enumerate(rows):
+            item = FacilityListItem.objects.create(
+                    row_index=idx,
+                    raw_data=','.join(row.raw_json.values()),
```

Comment:
I suppose that we need to consolidate the data that we pass to `raw_data` because in `facility_view_set.py` this field looks like `raw_data=json.dumps(request.data)`

Reply:
I think everything here is correct since `raw_data` is the stringified data of the row, not including the header. I believe the mistake is in the `facility_view_set.py` since the whole object is assigned to the `raw_data` of the facility list item, which seems incorrect because `raw_data` only stands for "_The full, unparsed CSV line as it appeared in the file._" whereas `raw_header` stands for "_The full, unparsed CSV header as it appeared in the file._"